packages/core/src/Stores/notification-store.js

```diff
@@ -0,0 +1,897 @@
+        this.notification_messages = this.notification_messages.filter(n => n.key !== key);
+        // Add notification messages to LocalStore when user closes, check for redundancy
+        const active_loginid = LocalStore.get('active_loginid');
+        if (!excluded_notifications.includes(key) && active_loginid) {
+            const messages = LocalStore.getObject('notification_messages');
+            // Check if same message already exists in LocalStore for this account
+            if (messages[active_loginid] && messages[active_loginid].includes(key)) {
+                return;
+            }
+            const current_message = () => {
+                if (Array.isArray(messages[active_loginid])) {
+                    messages[active_loginid].push(key);
+                    return messages[active_loginid];
+                }
+                return [key];
+            };
+            if (!should_show_again) {
+                // Store message into LocalStore upon closing message
+                Object.assign(messages, { [active_loginid]: current_message() });
+                LocalStore.setObject('notification_messages', messages);
+            }
+        }
+    }
+
+    @action.bound
+    removeNotificationMessageByKey({ key }) {
+        this.notification_messages = this.notification_messages.filter(n => n.key !== key);
+    }
+
+    @action.bound
+    resetVirtualBalanceNotification(loginid) {
+        const { accounts, is_logged_in } = this.root_store.client;
+        if (!is_logged_in) return;
+        if (!accounts[loginid].is_virtual) return;
+        const min_reset_limit = 1000;
+        const max_reset_limit = 999000;
+        const balance = parseInt(accounts[loginid].balance);
+
+        // Display notification message to user with virtual account to reset their balance
+        // if the balance is less than equals to 1000 or more than equals to 999000
+        if (balance <= min_reset_limit || balance >= max_reset_limit) {
+            let message = localize(
+                'Your demo account balance is low. Reset your balance to continue trading from your demo account.'
+            );
+            if (balance >= max_reset_limit)
+                message = localize(
+                    'Your demo account balance has reached the maximum limit, and you will not be able to place new trades. Reset your balance to continue trading from your demo account.'
+                );
+            this.setClientNotifications({ resetVirtualBalance: this.resetVirtualBalance, message });
+            this.addNotificationMessage(this.client_notifications.reset_virtual_balance);
+        } else {
+            this.removeNotificationByKey({ key: 'reset_virtual_balance' });
+            this.removeNotificationMessage({ key: 'reset_virtual_balance', should_show_again: true });
+        }
+    }
+
+    @action.bound
+    setClientNotifications(client = {}) {
+        // TODO: Update links to app_2 links when components are done. (Mahdi)
```

Comment:
I think we can remove this comment. Maybe there was a different plan at that time. It doesn't make any sense now.